### PR TITLE
Refactor about:preferences#payments followups

### LINKED ISSUE #3485: Payment History popover probably needs a little more attention
it seems sort of "drab" in comparison to the rest of the graphic treatments...

### LINKED ISSUE #5719: Close button is not aligned with text in payment history modal
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
Close button is not aligned with text in payment history window

**Expected behavior:**
Should be aligned with the text

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows 10 x64

- Brave Version:
0.12.8 (possibly on older versions too)

- Steps to reproduce:
    1. Click on view payment history in Payments 
    2. In the view payment history modal, close button is not aligned with the text 

- Screenshot if needed:
image

- Any related issues:
cc: @bradleyrichter @jkup

### LINKED ISSUE #5799: Addfunds dialog needs a little margin
as noted in https://github.com/brave/browser-laptop/pull/5797#issuecomment-262367178

446c2cfc-b0b5-11e6-83b1-9949cbecbcd3

### LINKED ISSUE #6048: Transfer funds button needs margin-bottom
**Describe the issue you encountered:** Transfer funds button (Transfer BTC) needs margin-bottom.

screenshot 2016-12-07 4 10 03

**Expected behavior:**

screenshot 2016-12-07 4 10 40

- Brave Version: master branch

- Any related issues: #6009 

I'm sending a quick PR.

### LINKED ISSUE #6061: Place the footer on the contribution report properly
**Describe the issue you encountered:**

https://github.com/brave/browser-laptop/issues/6039#issuecomment-265333789 courtesy of @mrose17 
> The OK button is not placed properly (see attachment)
>
> 

**Expected behavior:**
> place OK button properly

- Any related issues: #6039 #3485

### LINKED ISSUE #6081: the buttons on about:preferences#payments need more paddings
**Describe the issue you encountered:**

screenshot 2016-12-08 12 41 23

**Expected behavior:**

screenshot 2016-12-08 12 44 06

- Brave Version: master branch

- Any related issues: #6009

### LINKED ISSUE #6083: Funds FAQ icon is underlined on hover
**Describe the issue you encountered:**

screenshot 2016-12-08 14 02 48

**Expected behavior:** 

- Platform (Win7, 8, 10? macOS? Linux distro?): macOS

- Brave Version: master branch

- Any related issues: #6009

### LINKED ISSUE #6201: Design "hide sites if" options on Advanced Options Dialog
*This is a follow-up of #4681*

courtesy of @bradleyrichter 
image

I'm creating like this (the select element styling is based on #6084):
screenshot 2016-12-12 0 19 53

- Platform (Win7, 8, 10? macOS? Linux distro?):

- Brave Version: master branch

- Any related issues:

### LINKED ISSUE #6202: Design payment history table
courtesy of @bradleyrichter 

https://github.com/brave/browser-laptop/issues/5719#issuecomment-262431933

image

- Brave Version: master branch

- Any related issues:

### LINKED ISSUE #6216: Buttons on about:payments are thinner than those on other about pages
**Describe the issue you encountered:**
screenshot 2016-12-15 5 34 15screenshot 2016-12-15 5 36 50

**Expected behavior:**
screenshot 2016-12-15 5 38 13screenshot 2016-12-15 5 38 05

- Platform (Win7, 8, 10? macOS? Linux distro?): macOS

- Brave Version: master branch

- Steps to reproduce:
    1. Open about:preferences
    2. Open about:preferences#payments

- Any related issues: this should be fixed with #6047

### LINKED ISSUE #6219: The buttons on the footers of modal dialogs of payments need to be aligned to right
**Describe the issue you encountered:**
The buttons on the footers of modal dialogs of payments placed at the center
screenshot 2016-12-15 11 55 20

**Expected behavior:**
They need to be aligned to right based on https://github.com/brave/browser-laptop/issues/5719#issuecomment-261322404

screenshot 2016-12-15 12 11 00

- Brave Version: master branch

- Any related issues: this will be fixed with #6047

### LINKED ISSUE #6220: Recovery keys should not be line-wrapped
**Describe the issue you encountered:**
https://github.com/brave/browser-laptop/issues/4807#issuecomment-256075424
> when we do get around to addressing this issue, perhaps we can also deal with the line-wrap issue!
>
> 

**Expected behavior:**

screenshot 2016-12-15 12 35 58

- Brave Version: master branch

- Any related issues: this will be fixed with #6047

### LINKED ISSUE #6221: Title bar should not have strange spacing in general (on about:preferences#payments)
**Describe the issue you encountered:**
https://github.com/brave/browser-laptop/issues/5719#issuecomment-261322404

> - title bar has strange spacing in general

Example on master branch:
screenshot 2016-12-15 12 42 55

**Expected behavior:**
screenshot 2016-12-15 12 45 27

- Brave Version: master branch

- Any related issues: #6047

### LINKED ISSUE #6222: fa-mobile should be aligned at the center
**Describe the issue you encountered:**
On about:preferences#payments, click "add funds" to open the modal dialog.

fa-mobile is not aligned at the center
screenshot 2016-12-15 13 00 56

**Expected behavior:**
It should be aligned at the center like this:
screenshot 2016-12-15 13 01 09

- Brave Version: master branch

- Any related issues: #6047

### LINKED ISSUE #6223: border-radius of modal dialog in .paymentsContainer should be 8px
**Expected behavior:**
Per discussion with @bradleyrichter the border-radius of modal dialog in .paymentsContainer from 4px to 8px.

From:
screenshot 2016-12-15 13 19 16

To:
screenshot 2016-12-15 13 19 28

- Brave Version: master branch

- Any related issues: #6047

### LINKED ISSUE #6225: verified icon on the ledger table should be outside of the anchor link to the site
**Describe the issue you encountered:**
The verified icon on the ledger table is linked to the site, which is not expected (users would expect to confirm the status of verification with the icon). This happens because the icon which is actually included in the anchor link is moved with `position: absolute;`.

hover

**Expected behavior:**
verified icon on the ledger table should be outside of the anchor link to the site. 

- Brave Version: master branch

- Steps to reproduce:
    1. Visit https://clifton.io some times to display the verified icon on about:preferences#payments

- Any related issues: #6047

### LINKED ISSUE #6229: "Your Payment History" should be orange
**Describe the issue you encountered:**
"Your Payment History" is black.
screenshot 2016-12-15 18 13 19

**Expected behavior:**
"Your Payment History" should be orange according to https://github.com/brave/browser-laptop/issues/5719#issuecomment-262431933
screenshot 2016-12-15 18 14 51

- Brave Version: master branch

- Any related issues: #6047

### LINKED ISSUE #6230: receipt links should not be wrapped
**Describe the issue you encountered:**
On "Your Payment History" the receipt links are sometimes wrapped

screenshot 2016-12-15 18 19 28

**Expected behavior:**
They should not be wrapped
screenshot 2016-12-15 18 18 37

- Brave Version: master branch

- Any related issues: #6047

### LINKED ISSUE #6231: border-bottom of the table header of the payment history should be extended
**Describe the issue you encountered:**
According to https://github.com/brave/browser-laptop/issues/5719#issuecomment-262431933 border-bottom of the table header of the payment history should be extended to the the border of the dialog.

screenshot 2016-12-15 18 00 28

**Expected behavior:**
screenshot 2016-12-15 18 26 33

- Brave Version: master branch

- Any related issues: #6047

### LINKED ISSUE #6232: OK button should be larger on Payment History dialog
**Describe the issue you encountered:**
Per @bradleyrichter the OK button should be larger on Payment History dialog 

screenshot 2016-12-15 18 04 38

The footer placement issue was tracked at #6061 and will be fixed with #6047 

**Expected behavior:**

screenshot 2016-12-15 18 04 51

- Brave Version: master branch

- Any related issues: #6047

### PR DESCRIPTION
**Pull request by luixxiul**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

*This is a follow-up of #6009*

This PR has 6 commits:
1. Polishment of titleBar and walletBar
2. Made the selectors specific on about:preferences#payments to avoid unexpected regressions 
3. Added margin-bottom of the transfer funds button (Redo #6049)
4. Fixed advanced settings modal dialogs on about:preferences#payments
5. Polishment of modal dialog and advanced settings dialog on about:preferences#payments
6. Restyled payment history table

## 1st commit

Polishment of titleBar and walletBar

On about:preferences#payments
- Disabled underline of the anchor link of fundsFAQ (fixes #6083)
- Set font-size of the elements to 14.5px
- Set margin of the elements to 12px
- Set min-width to the select, input, and button
- Set padding, based on the margin of the items inside it

On about:preferences
- Made button size consistent by replacing em with rem (to avoid font-size inconsistency of the buttons inside walletBar) (fixes #6216)
- Set the default margin of the buttons to 0

screenshot 2016-12-10 14 31 04

### Test Plan 1

1. Open about:preferences
2. Make sure the style of about:preferences looks consistent

## 2nd commit

Made the selectors specific on about:preferences#payments to avoid unexpected regressions

- Classes and ids were renamed (fixes #6081)
- Classes were moved under .paymentsContainer, .bitcoinDashboard, and .ledgerTable
- Some elements and classes were removed
- Moved elements above classes/ids
- Moved .verified out of anchor link (the verified icon is no longer clickable) (fix #6225)

See the commit message for more details

Also .paymentHistoryFooter was introduced to fix #6061
screenshot 2016-12-10 14 41 44

### Test Plan 2

1. Make sure all of the selectors listed up in the commit message (except removed ones) can be actually found in the code, especially `preferences.js`
2. Make sure the style of about:preferences#payments is not broken
    - Open about:preferences#payments
    - Toggle off/on switch
    - Click "Advanced Settings..."
    - Click "Backup your wallet" and "Done"
    - Click "Recover your wallet" and "Done"
    - Click "Add funds..."
    - Click "Display QR code"
3. Make sure the verified icon is no longer clickable

## 3rd commit

Added margin-bottom of the transfer funds button (Redo #6049) 
This fixes #6048

screenshot 2016-12-10 14 42 30

### Test Plan 3

1. Open https://jsfiddle.net/LnwtLckc/5/
2. Click "register"
3. Make sure "Transfer BTC" button has the margin-bottom on add funds dialog on about:preferences#payments

## 4th commit

*This commit will be squashed with the 5th commit after review*

Fixed advanced settings modal dialogs on about:preferences#payments

Added white-space:nowrap to recovery keys (fixes #6220)

### Test Plan 4
(n/a; covered by Test Plan 5)

## 5th commit
Polishment of modal dialog and advanced settings dialog on about:preferences#payments

Mockup images by @bradleyrichter: https://github.com/brave/browser-laptop/issues/5719#issuecomment-262431933

Restyled modal dialog under .paymentsContainer
- Made the background of modal dialog consistent
- Changed border-radius of modal dialog (fixes #6223)
- Fixed spacing of the modal dialog header (fixes #6221)
- Set margin-top/margin-bottom to panel inside .board
- Aligned buttons in the footer with flex-end (fixes #6219)
- Aligned fa icons on the payment panel with display:flex (fixes #6222)
- Fixed panelFooter in .addFundsBoard (fixes #5799)
  - Aligned Coinbase icon and message
  - Display the footer

screenshot 2016-12-15 3 32 43

For US
screenshot 2016-12-15 3 25 51

For non-US
screenshot 2016-12-15 3 23 27

Restyled advanced settings dialog
- Designed "hide sites if" options on the dialog (closes #6201, which is a follow-up of #4681)
- Made the margin consistent between the headers/labels and the input/select elements
  - Addresses https://github.com/brave/browser-laptop/pull/6047#issuecomment-265946727

screenshot 2016-12-15 3 33 58

## Test Plan 5

For "Restyled modal dialog under .paymentsContainer"
1. Enable Coinbase if not
2. Open about:preferences#payments
3. Click "Add funds..."
4. Make sure the board is well styled
5. Make sure the footer is aligned horizontally
6. Click "Fund with debit/credit"
7. Make sure Coinbase widget is displayed with transparent background
8. Open https://jsfiddle.net/LnwtLckc/5/
9. Click "register" and go back
10. Make sure "Transfer BTC" button is aligned to right

For "Restyled advanced settings dialog"
1. Click "Advanced Settings..."
2. Check the margin between the labels and the select forms
3. Click "Backup your wallet"
4. Check the margin between the labels and the keys
5. Click "Done" and "Recover your wallet"
6. Check the margin between the labels and the input forms

## 6th commit
Restyled payment history table

Closes #6202
Closes #3485

- Removed "pull-right" class from the close button in modalOverlay.less (to confirm 21ece29 of #6009 fixed #5719)
- Added color:@braveMediumOrange to .sectionTitle of .paymentHistory (fixes #6229)
- Added padding-left and padding-right to th and td (fixes #6231)
- Added white-space:nowrap to disable wrap (fixes #6230)
- Added min-width:80px to the button inside the footer (fixes #6232)
- Made sectionTitle consistent
  - Removed ":not(.paymentHistory)" from modalOverlay.less

screenshot 2016-12-15 3 36 37

Test Plan:
1. Disable the ledger
2. run `npm run add-simulated-payment-history` some times
3. Enable the ledger
4. Click "View Payment History..."
5. Click "OK" button

### CODE REVIEW

**Comment by luixxiul** [comment]
@bsclifton will you please review? If it is OK, I will squash the last 6 commits into 61d075e. Thanks in advance.

File: less/about/preferences.less [hunk 1]
```diff
@@ -1134,17 +1144,6 @@ table.sortableTable {
       }
     }
 
-    span {
```
Comment (luixxiul) [inline less/about/preferences.less]: Note: this is outside of .walletBar. The commit message will be fixed when squashing.

**Comment by luixxiul** [comment]
`#paymentHistory` was replaced with `.paymentHistoryTable`.

File: less/about/preferences.less [hunk 2]
```diff
@@ -587,6 +509,261 @@ div.nextPaymentSubmission {
...
+            }
+          }
+        }
+
+        .dialog-footer {
+          background: #EEE;
+          padding-top: 10px;
+          padding-bottom: 10px;
+
+          .paymentHistoryFooter {
```
Comment (luixxiul) [inline less/about/preferences.less]: This will fix #6061

**Comment by luixxiul** [comment]
Rebased for now.

**Comment by bsclifton** [comment]
@luixxiul if you want to squash it up, I'm finally checking it out now :smile:

I'm not sure how many more UI changes we should pull in, so I'm leaving it as 0.13.1 for now... Will leave notes here after I get a chance to try it :smile:

**Comment by bsclifton** [comment]
Actually, with how bad https://github.com/brave/browser-laptop/issues/6061 looks, maybe this should be pulled in

**Comment by bsclifton** [comment]
For example, I see the title over the select boxes is very close, almost touching:
screen shot 2016-12-08 at 11 51 04 pm

Because of the problems we've had with our less being fragile (ex: the selectors not being specific enough), I'd have to try master and compare to know if that changed or not

**Comment by luixxiul** [comment]
@bsclifton thanks, I've noticed it from the first. Should I include the fix into this PR? It will be like this: https://github.com/luixxiul/browser-laptop/commit/2327e9eca53eb4a637dd4c3e3a2a1d1f613c2fbc#commitcomment-19937858

Image 6
screenshot 2016-11-23 22 27 22

Image 7
screenshot 2016-11-23 22 27 28

Image 8
screenshot 2016-11-23 22 27 36

**Comment by luixxiul** [comment]
@bsclifton rather should I add comments on lines https://github.com/brave/browser-laptop/pull/6047/files to clarify what I changed?

**Comment by bsclifton** [comment]
The change you made above, let's definitely include with this PR

The screenshots are ok- it would help to have a quick summary like:
- Payments > Advanced settings: moved up text above select fields (it was too close)

Actually- looking at your commit history, those messages are almost perfect :smile: The pictures are only useful (to me as a reviewer at least) if there is a before and after

**Comment by luixxiul** [comment]
@bsclifton Updated and squashed! I added the summary and the QA step for each commit in the original post of this PR.

**Comment by luixxiul** [comment]
I noticed the verified icon was not placed on the right place. Fixing.

**Comment by luixxiul** [comment]
Updated, removing `input[type='range']` (as it exists in sortable.less)

**Comment by luixxiul** [comment]
Rebased for now.

**Comment by luixxiul** [comment]
I added another two commits, and most of the polish work as far as I could for now was done with them. 

@bsclifton would you please let me know when the review is finished? I would like to sign the commits with my GPG key and rewrite commit messages to add "#" to to close issues/link to them automatically (because I edit the pushed commit so much the linked issues/PRs has a lot of notifications). Thanks!

**Comment by bsclifton** [comment]
@luixxiul sorry I haven't had time to look at this yet :frowning_face: 

@jkup would you be able to help me try this out and see how it looks?

**Comment by luixxiul** [comment]
@bsclifton nevermind, no more work on this PR will be added.

**Comment by luixxiul** [comment]
Updated, adding the tickets numbers without the hash.

**Comment by jkup** [comment]
Took a quick pass at this and it all looks good! I think for next time it would be a lot easier to review as 5 or 6 different tickets. Also there is a ton of CSS changes in this diff but it looks like most of it is just renaming id's to classes and indentation. @luixxiul is there anything particular I should be testing here?

Otherwise ++

**Comment by luixxiul** [comment]
@jkup thanks for reviewing! As I live in outside the US, I would like you to check if the buttons on the dialog which appears by clicking "add funds" on preferences#payments works as expected, especially "fund with debit/credit" (whether it successfully opens the Coinbase widget) and "display QR code" (whether the modal dialog has the three icons in the footer).

@bsclifton @jkup if both of you are ++, I would like to edit the commit message to link commits to issues and PR. Please let me know.

**Comment by bsclifton** [comment]
Big ++ from me too!  Really nice cleanup of the original post, BTW :smile: 

This was an intimidating pull request for sure- but the work is very much appreciated. Going forward, I don't think we'll ever have this many tweaks to the same pages again

Squash it up and then I'll give it one last test and can merge after that :smile:

**Comment by bsclifton** [comment]
Looks great, merging! :smile:

**Pull request merged**